processor/src/fast/mod.rs

```diff
@@ -684,6 +686,21 @@ impl FastProcessor {
     }
 }
 
+// EXECUTION OUTPUT
+// ===============================================================================================
+
+/// The output of a program execution, containing the state of the stack, advice provider, and
+/// memory at the end of the execution.
+#[derive(Debug)]
+pub struct ExecutionOutput {
+    pub stack_outputs: StackOutputs,
+    pub advice: AdviceProvider,
+    pub memory: Memory,
```

Comment:
nit: I would probably rename `stack_outputs` to just `stack` as the "outputs" part is implied by the struct name as well as field type. Also, this would be more consistent with other fields (e.g., we don't say `advice_outputs`).

Reply:
Done